### PR TITLE
Staleness and fault management fix

### LINKED ISSUE #434: Staleness, Fault management changed in realtime don't work
#### Summary
Staleness and fault management changes in realtime are not reflected in Open MCT

#### Expected vs Current Behavior

Staleness and Fault Management status should update in realtime.

#### Impact Check List

- [x] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?

#### Steps to Reproduce
1. Create an overlay plot with a telemetry endpoint like the `timelyParameter`
2. Ensure that staleness is setup for this telemetry endpoint (set timely parameter script needs to be for this)
3. Ensure that fault management is setup for this telemetry endpoint (not sure how to set this up)
4. Check that staleness indicators are displayed when data for the telemetry endpoint is stale.
5. Check that fault management works for this telemetry endpoint.

#### Environment
* Open MCT Version: Open MCT 4.0.0
* Deployment Type: dev
* OS: All
* Browser: all

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes #434 

### Describe your changes:
Use the reatime provider for subscriptions (not openmct)

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by shefalijoshi** [comment]
@unlikelyzero Please provide test setup for staleness and fault management for e2e testing

File: src/providers/fault-mgmt-providers/realtime-fault-provider.js [hunk 1]
```diff
@@ -34,13 +33,13 @@ export default class RealtimeFaultProvider {
     }
 
     subscribe(domainObject, callback) {
-        const globalUnsubscribe = this.#openmct.telemetry.subscribe(
+        const globalUnsubscribe = this.realtimeProvider.subscribe(
```
Comment (ozyx) [inline src/providers/fault-mgmt-providers/realtime-fault-provider.js:36]: why can't we use the openmct API methods here?

**Review by ozyx — CHANGES_REQUESTED** [review]

File: src/providers/fault-mgmt-providers/realtime-fault-provider.js (hunk 1 above)
Comment (shefalijoshi) [reply src/providers/fault-mgmt-providers/realtime-fault-provider.js:36]: The Open MCT API looks for the provider and calls the provider.subscribe method, The provider in this case is the realtime fault management provider. So pointing it back to the Open MCT API would not make any sense.

Comment (ozyx) [reply src/providers/fault-mgmt-providers/realtime-fault-provider.js:36]: That's how consumers are supposed to use the API though, right? We should be dogfooding where possible. @akhenry thoughts ?

**Review by ozyx — CHANGES_REQUESTED** [review]
Changes look good, let's just get some test stubs in there as these are likely going to be complex tests where we're trailblazing some new patterns

File: src/providers/staleness-provider.js [hunk 2]
```diff
@@ -24,9 +24,9 @@ import { OBJECT_TYPES, STALENESS_STATUS_MAP } from '../const.js';
 import { buildStalenessResponseObject } from '../utils.js';
 
 export default class YamcsStalenessProvider {
-    #openmct;
-    constructor(openmct, latestTelemetryProvider) {
-        this.#openmct = openmct;
+    constructor(openmct, realtimeTelemetryProvider, latestTelemetryProvider) {
```
Comment (ozyx) [inline src/providers/staleness-provider.js:27]: we should remove the `openmct` parameter as it's unused

File: tests/e2e/yamcs/faultManagement.e2e.spec.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ *****************************************************************************/
+
+/*
+Staleness Specific Tests
+*/
+
+import { test, expect } from '../opensource/pluginFixtures.js';
+import { createDomainObjectWithDefaults } from '../opensource/appActions.js';
+
+test.describe.skip("Fault management tests @yamcs", () => {
```
Comment (ozyx) [inline tests/e2e/yamcs/faultManagement.e2e.spec.js:30]: use `fixme` for test stubs
```suggestion
test.describe.fixme("Fault management tests @yamcs", () => {
```

File: tests/e2e/yamcs/staleness.e2e.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ *****************************************************************************/
+
+/*
+Staleness Specific Tests
+*/
+
+import { test, expect } from '../opensource/pluginFixtures.js';
+import { createDomainObjectWithDefaults } from '../opensource/appActions.js';
+
+test.describe.skip("Staleness tests @yamcs", () => {
```
Comment (ozyx) [inline tests/e2e/yamcs/staleness.e2e.js:30]:
```suggestion
test.describe.fixme("Staleness tests @yamcs", () => {
```

**Review by ozyx — CHANGES_REQUESTED** [review]
Couple more minor changes and good to go

File: src/providers/staleness-provider.js (hunk 2 above)
Comment (shefalijoshi) [reply src/providers/staleness-provider.js:27]: Good catch. done.

**Review by ozyx — APPROVED** [review]
LGTM

**Pull request merged**